views/js/previewer/plugins/tools/scale/scale.js

```diff
@@ -0,0 +1,532 @@
+         * @param {Event} event - event object related to affected control.
+         * @private
+         */
+        onOrientationChange: function onOrientationChange(event) {
+            var element = event.target;
+            var container = controls.$scaleWrapper.find('.' + selectedDevice + '-preview-container');
+            var newOrientation = element.value;
+
+            if (newOrientation === orientation) {
+                return false;
+            }
+            sizeSettings = {
+                height: container.width(),
+                width: container.height()
+            };
+
+            container.css(sizeSettings);
+
+            this.setOrientation(newOrientation);
+            this.scaleFrame();
+        },
+        
+        /**
+         * Configures selected device type on which depends all other parts of the plugin
+         * decides whether it needs to draw device frame around item content and scale it accordingly
+         * @param {Event} event - event object related to affected control.
+         * @private
+         */
+        onDeviceTypeChange: function onDeviceTypeChange(event) {
+            var element = event.target;
+
+            selectedDevice = element.value;
+
+            this.composeControlsByDeviceType(element.value);
+
+            if(element.value === DEFAULT_TYPE){
+                this.removeDeviceFrame();
+            }else{
+                this.changeDeviceFrame(element.value, true);
+                this.scaleFrame();
+                this.setupScreenSize();
+                if (element.value === 'mobile'){
+                    controls.$mobileDevices.children('.mobile-device-selector').get(0).dispatchEvent(new Event('change'));
+                } else if( element.value === 'desktop'){
+                    controls.$desktopDevices.get(0).dispatchEvent(new Event('change'));
+                }
+
+            }
+        }
+
+    };
+
+    return pluginFactory({
+
+        name: 'scale',
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init: function init() {
+            var self = this;
+            testRunner = this.getTestRunner();
+
+
+            controls = this.controls = {
+                $deviceTypes: $(previewTypesTpl({
+                    items: _getPreviewTypes(),
+                })),
+                $mobileDevices: $(mobileDevicesTpl({
+                    items: devices.byType('mobile'),
+                })),
+                $desktopDevices: $(desktopDevicesTpl({
+                    items: devices.byType('desktop'),
+                }))
+            };
+
+
+            /**
+             *  when user changes device type he/she want to test item in
+             *  @event scale#preview-scale-device-type
+             */
+            this.controls.$deviceTypes.on('change', function (event) {
+               api.onDeviceTypeChange(event);
```

Comment:
Relying on an external service to manage the behavior is not so bad, but it should not be a static object. You should rather use a factory, to get a proper instance, and bind the context to it or at least provide the context as a parameter.

However, since all the behavior is managed in the `init`, is it required to move that apart?